backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/checkbox/BpkCheckbox.kt

```diff
@@ -0,0 +1,167 @@
+import androidx.compose.foundation.selection.triStateToggleable
+import androidx.compose.material.CheckboxDefaults
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.LocalMinimumTouchTargetEnforcement
+import androidx.compose.material.TriStateCheckbox
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.invisibleToUser
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.state.ToggleableState
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.utils.BpkToggleableContent
+import net.skyscanner.backpack.compose.utils.applyIf
+
+@Composable
+fun BpkCheckbox(
+  text: String,
+  checked: Boolean,
+  onCheckedChange: ((Boolean) -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
+  BpkCheckbox(
+    checked = checked,
+    onCheckedChange = onCheckedChange,
+    interactionSource = interactionSource,
+    enabled = enabled,
+    modifier = modifier,
+    content = { BpkText(text) },
+  )
+}
+
+@Composable
+fun BpkCheckbox(
+  text: String,
+  state: ToggleableState,
+  onClick: (() -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
+  BpkCheckbox(
+    state = state,
+    onClick = onClick,
+    modifier = modifier,
+    enabled = enabled,
+    interactionSource = interactionSource,
+    content = { BpkText(text) },
+  )
+}
+
+@Composable
+fun BpkCheckbox(
+  checked: Boolean,
+  onCheckedChange: ((Boolean) -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+  content: @Composable RowScope.(Boolean) -> Unit,
+) {
+  BpkCheckbox(
+    state = ToggleableState(checked),
+    onClick = if (onCheckedChange != null) { { onCheckedChange(!checked) } } else null,
+    interactionSource = interactionSource,
+    enabled = enabled,
+    modifier = modifier,
+    content = { content(checked) },
+  )
+}
+
+@Composable
+fun BpkCheckbox(
+  state: ToggleableState,
+  onClick: (() -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+  content: @Composable RowScope.(ToggleableState) -> Unit,
+) {
+  Row(
+    verticalAlignment = Alignment.CenterVertically,
+    horizontalArrangement = Arrangement.SpaceBetween,
```

Comment:
you could use `Arrangement.spacedBy(BpkDimension.Spacing.Sm)` here and then avoid setting the padding end of the checkbox